napari/components/component.py

```diff
@@ -0,0 +1,10 @@
+from collections import defaultdict
+
+from contextlib import contextmanager
+
+
+class Component:
```

Comment:
What's the point of this class? There doesn't seem to be anything in it.

Reply:
We could get rid of this, i used to have some listener machinery here. But now we are using the events as you guys suggested, which turned out to be a good idea. We could get rid of this class, or keep it so that we eventually can factor up any common code between the components.

Reply:
i will delete now, can always add back in if / when we have a need